finance-admin/internal/server/upload.go

```diff
@@ -27,7 +28,12 @@ func (h *UploadHandler) render(v AppVars, w http.ResponseWriter, r *http.Request
 	}
 	defer file.Close()
 
-	data, err := shared.NewUpload(shared.ParseReportUploadType(reportUploadType), uploadDate, email, file, handler.Filename)
+	if handler.Filename != reportUploadType.Filename(uploadDate) && reportUploadType.Filename(uploadDate) != "" {
+		expectedFilename := strings.Replace(reportUploadType.Filename(uploadDate), ":", "/", -1)
```

Comment:
Why are the ":" being replaced with "/"?

Reply:
I'm not sure why this happens, but if the filename has a slash in it it's read by the filereader as a colon - I assume this is to differentiate it from the slash that would show that the file's in a directory. So, when the return the error to the user I'm just replacing the colons with slashes, otherwise it'd tell them to change the filename to feemoto_10:12:24normal.csv